src/Checkout/Payment/Method/PUIHandler.php

```diff
@@ -7,167 +7,93 @@
         }
 
-        try {
-            $this->puiCustomerDataService->checkForCustomerData($transaction->getOrder(), $dataBag, $salesChannelContext);
-            $this->settingsValidationService->validate($salesChannelContext->getSalesChannelId());
-            $this->orderTransactionStateHandler->process($transactionId, $salesChannelContext->getContext());
-
-            $order = $this->puiOrderBuilder->getOrder(
-                $transaction,
-                $salesChannelContext,
-                $dataBag,
-            );
+        $dataBag = new RequestDataBag($request->request->all());
+        $this->puiCustomerDataService->checkForCustomerData($transaction, $dataBag, $context);
 
-            $updateTime = $transaction->getOrderTransaction()->getUpdatedAt();
-
-            $paypalOrderResponse = $this->orderResource->create(
-                $order,
-                $salesChannelContext->getSalesChannelId(),
-                PartnerAttributionId::PAYPAL_PPCP,
-                true,
-                $transactionId . ($updateTime ? $updateTime->getTimestamp() : ''),
-                $fraudnetSessionId
-            );
-
-            $this->transactionDataService->setOrderId(
-                $transactionId,
-                $paypalOrderResponse->getId(),
-                PartnerAttributionId::PAYPAL_PPCP,
-                $salesChannelContext
-            );
-        } catch (PaymentException $e) {
-            if ($e->getParameter('orderTransactionId') === null && method_exists($e, 'setOrderTransactionId')) {
-                $e->setOrderTransactionId($transactionId);
-            }
-            throw $e;
-        } catch (\Exception $e) {
-            $this->logger->error($e->getMessage(), ['error' => $e]);
-
-            throw PaymentException::syncProcessInterrupted($transactionId, $e->getMessage());
-        }
+        return parent::pay($request, $transaction, $context, $validateStruct);
     }
 
-    /**
-     * @deprecated tag:v10.0.0 - Will be removed. Use PayPalController::handleError instead
-     */
-    public function handleError(PayPalApiException $exception): void
+    protected function isVaultable(): bool
     {
-        if ($exception->getStatusCode() !== Response::HTTP_UNPROCESSABLE_ENTITY) {
-            return;
-        }
+        return false;
+    }
 
-        $issue = $exception->getIssue();
-        if (!$issue) {
-            return;
-        }
+    protected function requirePreparedOrder(): bool
+    {
+        return false;
+    }
 
-        if (!\array_key_exists($issue, self::ERROR_KEYS)) {
-            return;
-        }
+    protected function executeOrder(PaymentTransactionStruct $transaction, Order $paypalOrder, OrderEntity $order, OrderTransactionEntity $orderTransaction, Context $context, bool $isUserPresent = true): Order
+    {
+        // Orders of APMs are created with `ORDER_COMPLETE_ON_PAYMENT_APPROVAL`
```

Comment:
```suggestion
        // Orders of PUI are created with `ORDER_COMPLETE_ON_PAYMENT_APPROVAL`
```

or is PUI an APM? :thinking: